crates/iceberg-catalog/src/api/management/v1/warehouse/mod.rs

```diff
@@ -676,7 +679,37 @@ pub trait Service<C: Catalog, A: Authorizer, S: SecretStore> {
             .await?;
         transaction.commit().await?;
 
-        // TODO: emit event
+        let num_tabulars = tabs.len();
+        let list_flags = ListFlags {
+            include_active: true,
+            include_staged: true,
+            include_deleted: true,
+        };
+        for (i, t) in tabs.iter().enumerate() {
+            if let Ok(Some(table_metadata)) =
+                C::get_table_metadata_by_id(warehouse_id, *t, list_flags, catalog.clone()).await
```

Comment:
IMO, we should change the undrop interface in the catalog trait to return the fields instead of calling the DB for every table we get here.